### PR TITLE
Initial Commit for backend models.

### PR DESCRIPTION
**Pull request by IsaacLance**

### CODE REVIEW

File: backend/blueprints/spa_api/service_layers/ml/advantage.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+import tensorflow as tf
+from tensorflow import keras
+from keras.models import load_model
+# Sklearn
+from sklearn.preprocessing import MinMaxScaler
+
+from carball.generated.api.game_pb2 import Game
+
+try:
+    from .config import ranges, model_path, droplist
```
Comment (Sciguymjm) [inline backend/blueprints/spa_api/service_layers/ml/advantage.py]: Try to avoid relative imports.

File: webapp/src/Components/Replay/ReplayViewer/Viewer.tsx [hunk 2]
```diff
@@ -124,9 +145,18 @@ export class Viewer extends Component<Props, State> {
...
-                    }
+                    },
                 })
             }
         )
     }
+
+    private readonly getMlData = () => {
+        const {replayId} = this.props
+        return Promise.all([doGet(`/replay/${replayId}/advantage`)])
```
Comment (Sciguymjm) [inline webapp/src/Components/Replay/ReplayViewer/Viewer.tsx]: No need for Promise.all here

File: webapp/src/Components/Replay/ReplayViewer/Viewer.tsx [hunk 3]
```diff
@@ -13,9 +13,12 @@ import {
...
 
+import { addFrameListener, FrameEvent } from "replay-viewer/eventbus/events/frame"
+import { doGet } from "../../../apiHandler/apiHandler"
 import { Replay } from "../../../Models"
 import { getReplayMetadata, getReplayViewerData, getReplayViewerDataRange } from "../../../Requests/Replay"
 import { LoadableWrapper } from "../../Shared/LoadableWrapper"
+import { AdvantageBarChart } from "./AdvantageBar"
 
 interface Props {
     replayId: Replay["id"]
```
Comment (Sciguymjm) [inline webapp/src/Components/Replay/ReplayViewer/Viewer.tsx]: Can we add a prop for making advantage optional? i.e. if it's not a 1s replay don't download the predictions.

File: webapp/src/Components/Replay/ReplayViewer/Viewer.tsx [hunk 4]
```diff
@@ -124,9 +145,18 @@ export class Viewer extends Component<Props, State> {
...
+                    },
                 })
             }
         )
     }
+
+    private readonly getMlData = () => {
+        const {replayId} = this.props
+        return Promise.all([doGet(`/replay/${replayId}/advantage`)])
+            .then(([advantage]) => {
```
Comment (Sciguymjm) [inline webapp/src/Components/Replay/ReplayViewer/Viewer.tsx]: If not Promise.all you don't need the `[]`
